# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/tests/browser/browser_hasActivePeerConnections.js
# Commit: 13a9d145a550
# Full Hash: 13a9d145a55053585dd7477bdc8b7842606784a6
# Author: Toshihito Kikuchi <tkikuchi@mozilla.com>
# Date: 2021-12-17 03:51:53
# Description:
#   Bug 1743663 - Remove windowContext.hasActivePeerConnections.  r=peterv
#   
#   TabUnloader.jsm can directly call windowGlobal.hasActivePeerConnections(),
#   so windowContext.hasActivePeerConnections is not needed.
#   
# ==============================================================================

diff -r 630013ad7f37 -r 13a9d145a550 dom/tests/browser/browser_hasActivePeerConnections.js
--- a/dom/tests/browser/browser_hasActivePeerConnections.js	Thu Dec 16 15:44:15 2021 +0000
+++ b/dom/tests/browser/browser_hasActivePeerConnections.js	Thu Dec 16 16:14:13 2021 +0000
@@ -8,11 +8,11 @@
 
 add_task(async () => {
   await BrowserTestUtils.withNewTab(TEST_URI1, async browser => {
-    const topWindow = browser.browsingContext.topWindowContext;
-    Assert.ok(topWindow);
+    const windowGlobal = browser.browsingContext.currentWindowGlobal;
+    Assert.ok(windowGlobal);
 
     Assert.strictEqual(
-      topWindow.hasActivePeerConnections,
+      windowGlobal.hasActivePeerConnections(),
       false,
       "No active connections at the beginning"
     );
@@ -30,7 +30,7 @@
     });
 
     Assert.strictEqual(
-      topWindow.hasActivePeerConnections,
+      windowGlobal.hasActivePeerConnections(),
       true,
       "One connection in the top window"
     );
@@ -48,7 +48,7 @@
     });
 
     Assert.strictEqual(
-      topWindow.hasActivePeerConnections,
+      windowGlobal.hasActivePeerConnections(),
       false,
       "All connections have been closed"
     );
@@ -106,7 +106,7 @@
     );
 
     Assert.strictEqual(
-      topWindow.hasActivePeerConnections,
+      windowGlobal.hasActivePeerConnections(),
       true,
       "#iframe-cross-origin still has an active connection"
     );
@@ -126,7 +126,7 @@
     });
 
     Assert.strictEqual(
-      topWindow.hasActivePeerConnections,
+      windowGlobal.hasActivePeerConnections(),
       false,
       "All connections have been closed"
     );
